compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1543,6 +1635,7 @@ public void visit(BLangErrorDestructure errorDeStmt) {
             if (varRef.cause.getKind() != NodeKind.SIMPLE_VARIABLE_REF ||
                     names.fromIdNode(((BLangSimpleVarRef) varRef.cause).variableName) != Names.IGNORE) {
                 setTypeOfVarRefInErrorBindingAssignment(varRef.cause);
+                checkInvalidTypeDef(varRef.cause);
```

Comment:
What about the detail bindings? Probably happens via org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer#checkErrorVarRefEquivalency?